### ISSUE TITLE
训练参数与模型不一致

### ISSUE THREAD

**Issue by yun-zero** [PaddlePaddle/PaddleOCR#11723]
- 系统环境/System Environment：
- mac
- 版本号/Version：2.5
- 运行指令/Command Code：
- python3 tools/train.py -c configs/rec/PP-OCRv3/ch_PP-OCRv3_rec_distillation.yml -o Global.pretrained_model=./pretrain_models/ch_PP-OCRv3_rec_train/best_accuracy
- 完整报错/Complete Error Message：
[2024/03/12 17:29:00] ppocr INFO: Architecture : 
[2024/03/12 17:29:00] ppocr INFO:     Models : 
[2024/03/12 17:29:00] ppocr INFO:         Student : 
[2024/03/12 17:29:00] ppocr INFO:             Backbone : 
[2024/03/12 17:29:00] ppocr INFO:                 last_conv_stride : [1, 2]
[2024/03/12 17:29:00] ppocr INFO:                 last_pool_type : avg
[2024/03/12 17:29:00] ppocr INFO:                 name : MobileNetV1Enhance
[2024/03/12 17:29:00] ppocr INFO:                 scale : 0.5
[2024/03/12 17:29:00] ppocr INFO:             Head : 
[2024/03/12 17:29:00] ppocr INFO:                 CTCHead : 
[2024/03/12 17:29:00] ppocr INFO:                     fc : 
[2024/03/12 17:29:00] ppocr INFO:                         weight : 6625
[2024/03/12 17:29:00] ppocr INFO:                 head_list : 
[2024/03/12 17:29:00] ppocr INFO:                     CTCHead : 
[2024/03/12 17:29:00] ppocr INFO:                         Head : 
[2024/03/12 17:29:00] ppocr INFO:                             fc_decay : 1e-05
[2024/03/12 17:29:00] ppocr INFO:                         Neck : 
[2024/03/12 17:29:00] ppocr INFO:                             depth : 2
[2024/03/12 17:29:00] ppocr INFO:                             dims : 64
[2024/03/12 17:29:00] ppocr INFO:                             hidden_dims : 120
[2024/03/12 17:29:00] ppocr INFO:                             name : svtr
[2024/03/12 17:29:00] ppocr INFO:                             use_guide : True
[2024/03/12 17:29:00] ppocr INFO:                     SARHead : 
[2024/03/12 17:29:00] ppocr INFO:                         enc_dim : 512
[2024/03/12 17:29:00] ppocr INFO:                         max_text_length : 6625
[2024/03/12 17:29:00] ppocr INFO:                 name : MultiHead
[2024/03/12 17:29:00] ppocr INFO:             Transform : None
[2024/03/12 17:29:00] ppocr INFO:             algorithm : SVTR
[2024/03/12 17:29:00] ppocr INFO:             freeze_params : False
[2024/03/12 17:29:00] ppocr INFO:             model_type : rec
[2024/03/12 17:29:00] ppocr INFO:             pretrained : None
[2024/03/12 17:29:00] ppocr INFO:             return_all_feats : True
[2024/03/12 17:29:00] ppocr INFO:         Teacher : 
[2024/03/12 17:29:00] ppocr INFO:             Backbone : 
[2024/03/12 17:29:00] ppocr INFO:                 last_conv_stride : [1, 2]
[2024/03/12 17:29:00] ppocr INFO:                 last_pool_type : avg
[2024/03/12 17:29:00] ppocr INFO:                 name : MobileNetV1Enhance
[2024/03/12 17:29:00] ppocr INFO:                 scale : 0.5
[2024/03/12 17:29:00] ppocr INFO:             Head : 
[2024/03/12 17:29:00] ppocr INFO:                 head_list : 
[2024/03/12 17:29:00] ppocr INFO:                     CTCHead : 
[2024/03/12 17:29:00] ppocr INFO:                         Head : 
[2024/03/12 17:29:00] ppocr INFO:                             fc_decay : 1e-05
[2024/03/12 17:29:00] ppocr INFO:                         Neck : 
[2024/03/12 17:29:00] ppocr INFO:                             depth : 2
[2024/03/12 17:29:00] ppocr INFO:                             dims : 64
[2024/03/12 17:29:00] ppocr INFO:                             hidden_dims : 120
[2024/03/12 17:29:00] ppocr INFO:                             name : svtr
[2024/03/12 17:29:00] ppocr INFO:                             use_guide : True
[2024/03/12 17:29:00] ppocr INFO:                     SARHead : 
[2024/03/12 17:29:00] ppocr INFO:                         enc_dim : 512
[2024/03/12 17:29:00] ppocr INFO:                         max_text_length : 6625
[2024/03/12 17:29:00] ppocr INFO:                 name : MultiHead
[2024/03/12 17:29:00] ppocr INFO:             Transform : None
[2024/03/12 17:29:00] ppocr INFO:             algorithm : SVTR
[2024/03/12 17:29:00] ppocr INFO:             freeze_params : False
[2024/03/12 17:29:00] ppocr INFO:             model_type : rec
[2024/03/12 17:29:00] ppocr INFO:             pretrained : None
[2024/03/12 17:29:00] ppocr INFO:             return_all_feats : True
[2024/03/12 17:29:00] ppocr INFO:     algorithm : Distillation
[2024/03/12 17:29:00] ppocr INFO:     model_type : rec
[2024/03/12 17:29:00] ppocr INFO:     name : DistillationModel
[2024/03/12 17:29:00] ppocr INFO: Eval : 
[2024/03/12 17:29:00] ppocr INFO:     dataset : 
[2024/03/12 17:29:00] ppocr INFO:         data_dir : ../DigitalTubeImage
[2024/03/12 17:29:00] ppocr INFO:         label_file_list : ['../DigitalTubeImage/Label.txt']
[2024/03/12 17:29:00] ppocr INFO:         name : SimpleDataSet
[2024/03/12 17:29:00] ppocr INFO:         transforms : 
[2024/03/12 17:29:00] ppocr INFO:             DecodeImage : 
[2024/03/12 17:29:00] ppocr INFO:                 channel_first : False
[2024/03/12 17:29:00] ppocr INFO:                 img_mode : BGR
[2024/03/12 17:29:00] ppocr INFO:             MultiLabelEncode : None
[2024/03/12 17:29:00] ppocr INFO:             RecResizeImg : 
[2024/03/12 17:29:00] ppocr INFO:                 image_shape : [3, 48, 320]
[2024/03/12 17:29:00] ppocr INFO:             KeepKeys : 
[2024/03/12 17:29:00] ppocr INFO:                 keep_keys : ['image', 'label_ctc', 'label_sar', 'length', 'valid_ratio']
[2024/03/12 17:29:00] ppocr INFO:     loader : 
[2024/03/12 17:29:00] ppocr INFO:         batch_size_per_card : 128
[2024/03/12 17:29:00] ppocr INFO:         drop_last : False
[2024/03/12 17:29:00] ppocr INFO:         num_workers : 4
[2024/03/12 17:29:00] ppocr INFO:         shuffle : False
[2024/03/12 17:29:00] ppocr INFO: Global : 
[2024/03/12 17:29:00] ppocr INFO:     cal_metric_during_train : True
[2024/03/12 17:29:00] ppocr INFO:     character_dict_path : ../PaddleOCR/pretrain_models/ch_PP-OCRv3_rec_train/digital_tube_dict.txt
[2024/03/12 17:29:00] ppocr INFO:     checkpoints : None
[2024/03/12 17:29:00] ppocr INFO:     debug : False
[2024/03/12 17:29:00] ppocr INFO:     distributed : False
[2024/03/12 17:29:00] ppocr INFO:     epoch_num : 800
[2024/03/12 17:29:00] ppocr INFO:     eval_batch_step : [0, 2000]
[2024/03/12 17:29:00] ppocr INFO:     infer_img : ../DigitalTubeImage/023471.jpg
[2024/03/12 17:29:00] ppocr INFO:     infer_mode : False
[2024/03/12 17:29:00] ppocr INFO:     log_smooth_window : 20
[2024/03/12 17:29:00] ppocr INFO:     max_text_length : 6625
[2024/03/12 17:29:00] ppocr INFO:     pretrained_model : ./pretrain_models/ch_PP-OCRv3_rec_train/best_accuracy
[2024/03/12 17:29:00] ppocr INFO:     print_batch_step : 10
[2024/03/12 17:29:00] ppocr INFO:     save_epoch_step : 3
[2024/03/12 17:29:00] ppocr INFO:     save_inference_dir : None
[2024/03/12 17:29:00] ppocr INFO:     save_model_dir : ./output/rec_ppocr_v3_distillation
[2024/03/12 17:29:00] ppocr INFO:     save_res_path : ./output/rec/predicts_ppocrv3_distillation.txt
[2024/03/12 17:29:00] ppocr INFO:     use_gpu : False
[2024/03/12 17:29:00] ppocr INFO:     use_space_char : False
[2024/03/12 17:29:00] ppocr INFO:     use_visualdl : False
[2024/03/12 17:29:00] ppocr INFO: Loss : 
[2024/03/12 17:29:00] ppocr INFO:     loss_config_list : 
[2024/03/12 17:29:00] ppocr INFO:         DistillationDMLLoss : 
[2024/03/12 17:29:00] ppocr INFO:             act : softmax
[2024/03/12 17:29:00] ppocr INFO:             dis_head : ctc
[2024/03/12 17:29:00] ppocr INFO:             key : head_out
[2024/03/12 17:29:00] ppocr INFO:             model_name_pairs : [['Student', 'Teacher']]
[2024/03/12 17:29:00] ppocr INFO:             multi_head : True
[2024/03/12 17:29:00] ppocr INFO:             name : dml_ctc
[2024/03/12 17:29:00] ppocr INFO:             use_log : True
[2024/03/12 17:29:00] ppocr INFO:             weight : 1.0
[2024/03/12 17:29:00] ppocr INFO:         DistillationDMLLoss : 
[2024/03/12 17:29:00] ppocr INFO:             act : softmax
[2024/03/12 17:29:00] ppocr INFO:             dis_head : sar
[2024/03/12 17:29:00] ppocr INFO:             key : head_out
[2024/03/12 17:29:00] ppocr INFO:             model_name_pairs : [['Student', 'Teacher']]
[2024/03/12 17:29:00] ppocr INFO:             multi_head : True
[2024/03/12 17:29:00] ppocr INFO:             name : dml_sar
[2024/03/12 17:29:00] ppocr INFO:             use_log : True
[2024/03/12 17:29:00] ppocr INFO:             weight : 0.5
[2024/03/12 17:29:00] ppocr INFO:         DistillationDistanceLoss : 
[2024/03/12 17:29:00] ppocr INFO:             key : backbone_out
[2024/03/12 17:29:00] ppocr INFO:             mode : l2
[2024/03/12 17:29:00] ppocr INFO:             model_name_pairs : [['Student', 'Teacher']]
[2024/03/12 17:29:00] ppocr INFO:             weight : 1.0
[2024/03/12 17:29:00] ppocr INFO:         DistillationCTCLoss : 
[2024/03/12 17:29:00] ppocr INFO:             key : head_out
[2024/03/12 17:29:00] ppocr INFO:             model_name_list : ['Student', 'Teacher']
[2024/03/12 17:29:00] ppocr INFO:             multi_head : True
[2024/03/12 17:29:00] ppocr INFO:             weight : 1.0
[2024/03/12 17:29:00] ppocr INFO:         DistillationSARLoss : 
[2024/03/12 17:29:00] ppocr INFO:             key : head_out
[2024/03/12 17:29:00] ppocr INFO:             model_name_list : ['Student', 'Teacher']
[2024/03/12 17:29:00] ppocr INFO:             multi_head : True
[2024/03/12 17:29:00] ppocr INFO:             weight : 1.0
[2024/03/12 17:29:00] ppocr INFO:     name : CombinedLoss
[2024/03/12 17:29:00] ppocr INFO: Metric : 
[2024/03/12 17:29:00] ppocr INFO:     base_metric_name : RecMetric
[2024/03/12 17:29:00] ppocr INFO:     ignore_space : False
[2024/03/12 17:29:00] ppocr INFO:     key : Student
[2024/03/12 17:29:00] ppocr INFO:     main_indicator : acc
[2024/03/12 17:29:00] ppocr INFO:     name : DistillationMetric
[2024/03/12 17:29:00] ppocr INFO: Optimizer : 
[2024/03/12 17:29:00] ppocr INFO:     beta1 : 0.9
[2024/03/12 17:29:00] ppocr INFO:     beta2 : 0.999
[2024/03/12 17:29:00] ppocr INFO:     lr : 
[2024/03/12 17:29:00] ppocr INFO:         learning_rate : 0.001
[2024/03/12 17:29:00] ppocr INFO:         name : Cosine
[2024/03/12 17:29:00] ppocr INFO:     name : Adam
[2024/03/12 17:29:00] ppocr INFO:     regularizer : 
[2024/03/12 17:29:00] ppocr INFO:         factor : 3e-05
[2024/03/12 17:29:00] ppocr INFO:         name : L2
[2024/03/12 17:29:00] ppocr INFO: PostProcess : 
[2024/03/12 17:29:00] ppocr INFO:     key : head_out
[2024/03/12 17:29:00] ppocr INFO:     model_name : ['Student', 'Teacher']
[2024/03/12 17:29:00] ppocr INFO:     multi_head : True
[2024/03/12 17:29:00] ppocr INFO:     name : DistillationCTCLabelDecode
[2024/03/12 17:29:00] ppocr INFO: Train : 
[2024/03/12 17:29:00] ppocr INFO:     dataset : 
[2024/03/12 17:29:00] ppocr INFO:         data_dir : ../DigitalTubeImage
[2024/03/12 17:29:00] ppocr INFO:         ext_op_transform_idx : 1
[2024/03/12 17:29:00] ppocr INFO:         label_file_list : ['../DigitalTubeImage/Label.txt']
[2024/03/12 17:29:00] ppocr INFO:         name : SimpleDataSet
[2024/03/12 17:29:00] ppocr INFO:         transforms : 
[2024/03/12 17:29:00] ppocr INFO:             DecodeImage : 
[2024/03/12 17:29:00] ppocr INFO:                 channel_first : False
[2024/03/12 17:29:00] ppocr INFO:                 img_mode : BGR
[2024/03/12 17:29:00] ppocr INFO:             RecConAug : 
[2024/03/12 17:29:00] ppocr INFO:                 ext_data_num : 2
[2024/03/12 17:29:00] ppocr INFO:                 image_shape : [48, 320, 3]
[2024/03/12 17:29:00] ppocr INFO:                 prob : 0.5
[2024/03/12 17:29:00] ppocr INFO:             RecAug : None
[2024/03/12 17:29:00] ppocr INFO:             MultiLabelEncode : None
[2024/03/12 17:29:00] ppocr INFO:             RecResizeImg : 
[2024/03/12 17:29:00] ppocr INFO:                 image_shape : [3, 48, 320]
[2024/03/12 17:29:00] ppocr INFO:             KeepKeys : 
[2024/03/12 17:29:00] ppocr INFO:                 keep_keys : ['image', 'label_ctc', 'label_sar', 'length', 'valid_ratio']
[2024/03/12 17:29:00] ppocr INFO:     loader : 
[2024/03/12 17:29:00] ppocr INFO:         batch_size_per_card : 128
[2024/03/12 17:29:00] ppocr INFO:         drop_last : True
[2024/03/12 17:29:00] ppocr INFO:         num_workers : 4
[2024/03/12 17:29:00] ppocr INFO:         shuffle : True
[2024/03/12 17:29:00] ppocr INFO: profiler_options : None
[2024/03/12 17:29:00] ppocr INFO: train with paddle 2.5.2 and device Place(cpu)
[2024/03/12 17:29:00] ppocr INFO: Initialize indexs of datasets:['../DigitalTubeImage/Label.txt']
[2024/03/12 17:29:00] ppocr INFO: Initialize indexs of datasets:['../DigitalTubeImage/Label.txt']
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Teacher.head.ctc_head.fc.weight [64, 15] not matched with loaded params Teacher.head.ctc_head.fc.weight [64, 6625] !
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Teacher.head.ctc_head.fc.bias [15] not matched with loaded params Teacher.head.ctc_head.fc.bias [6625] !
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Teacher.head.sar_head.decoder.embedding.weight [17, 512] not matched with loaded params Teacher.head.sar_head.decoder.embedding.weight [6627, 512] !
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Teacher.head.sar_head.decoder.prediction.weight [1536, 16] not matched with loaded params Teacher.head.sar_head.decoder.prediction.weight [1536, 6626] !
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Teacher.head.sar_head.decoder.prediction.bias [16] not matched with loaded params Teacher.head.sar_head.decoder.prediction.bias [6626] !
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Student.head.ctc_head.fc.weight [64, 15] not matched with loaded params Student.head.ctc_head.fc.weight [64, 6625] !
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Student.head.ctc_head.fc.bias [15] not matched with loaded params Student.head.ctc_head.fc.bias [6625] !
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Student.head.sar_head.decoder.embedding.weight [17, 512] not matched with loaded params Student.head.sar_head.decoder.embedding.weight [6627, 512] !
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Student.head.sar_head.decoder.prediction.weight [1536, 16] not matched with loaded params Student.head.sar_head.decoder.prediction.weight [1536, 6626] !
[2024/03/12 17:29:01] ppocr WARNING: The shape of model params Student.head.sar_head.decoder.prediction.bias [16] not matched with loaded params Student.head.sar_head.decoder.prediction.bias [6626] !
[2024/03/12 17:29:01] ppocr INFO: load pretrain successful from ./pretrain_models/ch_PP-OCRv3_rec_train/best_accuracy
[2024/03/12 17:29:01] ppocr INFO: train dataloader has 1 iters
[2024/03/12 17:29:01] ppocr INFO: valid dataloader has 2 iters
[2024/03/12 17:29:01] ppocr INFO: During the training process, after the 0th iteration, an evaluation is run every 2000 iterations

从日志中大概可以看出是训练的参数与模型不一致，但我是跟着教程一起做的，训练模型用的是https://paddleocr.bj.bcebos.com/PP-OCRv3/chinese/ch_PP-OCRv3_rec_train.tar。训练的配置文件是[ch_PP-OCRv3_rec_distillation.yml](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.7/configs/rec/PP-OCRv3/ch_PP-OCRv3_rec_distillation.yml)训练参数在上面日志中有。请问是否改更换模型还是修改配置参数

**Comment by yun-zero** [comment]
明白了，是字典的字符数和模型的不一致，请问对训练有什么影响吗

**Comment by TingquanGao** [comment]
这是因为加载的预训练模型的分类头和字典分类数不一致。只要确认字典是正确的，就没问题。

**Comment by TingquanGao** [comment]
长时间未回复，已close，如有问题，可reopen并补充
